ludwig/data/preprocessing.py

```diff
@@ -1173,8 +1174,12 @@ def precompute_fill_value(dataset_df, feature, preprocessing_parameters):
                 'Filling missing values with mean is supported '
                 'only for numerical types',
             )
-        return dataset_df[feature[COLUMN]].mean()
-
+        if isinstance(dataset_df, pd.DataFrame):
```

Comment:
Good catch! To make this more generic, what we can do is pass in the `backend` param from where this function is called in `build_metadata()`. So we can change the signature here to 

```
precompute_fill_value(dataset_df, feature, preprocessing_parameters, backend)
```

Then to get the mean, we just need to call:

```
return backend.df_engine.compute(dataset_df[feature[COLUMN]].mean())
```

Then we can remove the import of Dask at the top (since it is not a required package).

Reply:
Made the changes! thanks!